Modules/_io/stringio.c

```diff
@@ -463,22 +463,24 @@ _io_StringIO_truncate_impl(stringio *self, Py_ssize_t size)
 
 /*[clinic input]
 _io.StringIO.seek
-    pos: Py_ssize_t
-    whence: int = 0
+    offset as pos: Py_ssize_t
+    whence: int(c_default='0') = io.SEEK_SET
```

Comment:
You should also copy parameter docstrings. Otherwise the function docstring will look incomplete.

Reply:
The param docstrings are rendered to be a part of the docstring during clinic input parsing. It will be part of the whole docstring after `make clinic`. The subclasses will inherit this, unless they override the docstring explicitly.

Reply:
No. Subclasses "inherit" docstrings only if they do not have their own docstrings.

I put "inherit" in quotes because actually they do not do this. It is pydoc who falls back to looking up docstrings in the parent classes.

Look at the Argument Clinic generated docstrings and you will see that they are incomplete.

Reply:
<details>

```
Seek docs for <class '_io.BufferedRWPair'>:
BufferedRWPair.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

BufferedRWPair.__doc__:
 Change the stream position and return the new absolute position.

  offset
    Offset as byte count.
    Relative to the position given by 'whence'.
  whence
    Relative position, used by 'offset'. Valid values are:
    * io.SEEK_SET -- Start of stream (the default)
    * io.SEEK_CUR -- Current position
    * io.SEEK_END -- End of stream

Offsets relative to the start of the stream are usually zero or
positive, offsets relative to the current stream position may be
zero, positive or negative, and offsets relative to the end of the
stream are usually zero or negative.

Some platforms allow seeking beyond the end of a file.

Note that not all file objects are seekable.

Seek docs for <class '_io.BufferedRandom'>:
BufferedRandom.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

BufferedRandom.__doc__:
 None

Seek docs for <class '_io.BufferedReader'>:
BufferedReader.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

BufferedReader.__doc__:
 None

Seek docs for <class '_io.BufferedWriter'>:
BufferedWriter.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

BufferedWriter.__doc__:
 None

Seek docs for <class '_io.BytesIO'>:
BytesIO.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

BytesIO.__doc__:
 Change stream position and return the new absolute position.

Set the byte offset 'offset', relative to position indicated by 'whence':

Seek docs for <class '_io.FileIO'>:
FileIO.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

FileIO.__doc__:
 None

Seek docs for <class '_io.StringIO'>:
StringIO.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

StringIO.__doc__:
 Change the stream position and return the new absolute position.

Offsets relative to the start of the stream are usually zero or positive.
Offsets relative to the current and end of file positions must zero;
any other values are unsupported, and will raise OSError.

Some platforms allow seeking beyond the end of a file.

Note that not all file objects are seekable.

Seek docs for <class '_io.TextIOWrapper'>:
TextIOWrapper.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

TextIOWrapper.__doc__:
 Change the file position and return the new absolute position.

Offsets relative to the start of the file are usually zero or positive.
Offsets relative to the current and end of file positions must zero;
any other values are unsupported, and will raise the
io.UnsupportedOperation exception.

Some platforms allow seeking beyond the end of a file.

Note that not all file objects are seekable.

Seek docs for <class '_io._BufferedIOBase'>:
_BufferedIOBase.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

_BufferedIOBase.__doc__:
 Change the stream position and return the new absolute position.

  offset
    Offset as byte count.
    Relative to the position given by 'whence'.
  whence
    Relative position, used by 'offset'. Valid values are:
    * io.SEEK_SET -- Start of stream (the default)
    * io.SEEK_CUR -- Current position
    * io.SEEK_END -- End of stream

Offsets relative to the start of the stream are usually zero or
positive, offsets relative to the current stream position may be
zero, positive or negative, and offsets relative to the end of the
stream are usually zero or negative.

Some platforms allow seeking beyond the end of a file.

Note that not all file objects are seekable.

Seek docs for <class '_io._IOBase'>:
_IOBase.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

_IOBase.__doc__:
 Change the stream position and return the new absolute position.

  offset
    Offset as byte count.
    Relative to the position given by 'whence'.
  whence
    Relative position, used by 'offset'. Valid values are:
    * io.SEEK_SET -- Start of stream (the default)
    * io.SEEK_CUR -- Current position
    * io.SEEK_END -- End of stream

Offsets relative to the start of the stream are usually zero or
positive, offsets relative to the current stream position may be
zero, positive or negative, and offsets relative to the end of the
stream are usually zero or negative.

Some platforms allow seeking beyond the end of a file.

Note that not all file objects are seekable.

Seek docs for <class '_io._RawIOBase'>:
_RawIOBase.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

_RawIOBase.__doc__:
 Change the stream position and return the new absolute position.

  offset
    Offset as byte count.
    Relative to the position given by 'whence'.
  whence
    Relative position, used by 'offset'. Valid values are:
    * io.SEEK_SET -- Start of stream (the default)
    * io.SEEK_CUR -- Current position
    * io.SEEK_END -- End of stream

Offsets relative to the start of the stream are usually zero or
positive, offsets relative to the current stream position may be
zero, positive or negative, and offsets relative to the end of the
stream are usually zero or negative.

Some platforms allow seeking beyond the end of a file.

Note that not all file objects are seekable.

Seek docs for <class '_io._TextIOBase'>:
_TextIOBase.__text_signature__:
($self, offset, whence=io.SEEK_SET, /)

_TextIOBase.__doc__:
 Change the stream position and return the new absolute position.

  offset
    Offset as byte count.
    Relative to the position given by 'whence'.
  whence
    Relative position, used by 'offset'. Valid values are:
    * io.SEEK_SET -- Start of stream (the default)
    * io.SEEK_CUR -- Current position
    * io.SEEK_END -- End of stream

Offsets relative to the start of the stream are usually zero or
positive, offsets relative to the current stream position may be
zero, positive or negative, and offsets relative to the end of the
stream are usually zero or negative.

Some platforms allow seeking beyond the end of a file.

Note that not all file objects are seekable.


```

</details>

Reply:
> Look at the Argument Clinic generated docstrings and you will see that they are incomplete.

Of course. Only the `_IOBase` docstring will be generated completely with param docstrings inlined, as the PR stands now.

Reply:
Perhaps param docstrings are not a good fit for these functions, though.

Reply:
All docstrings should be complete. If possible, make them empty, then pydoc will show the docstring from the parent class. Perhaps you only need two docstrings: for binary and text streams.